code.js

```diff
@@ -1,3 +1,54 @@
 function dijkstra(graph, sourceNode) {
-    return [];
+   var distance = [];
+   var visited = [];
+   var remaining = true;
+   var current = sourceNode;
+
+   for (var i = 0; i < graph.length; i++) {
+      distance[i] = Infinity;
+      visited[i] = false;
+   }
+
+   distance[sourceNode] = 0;
+
+   while(remaining) {
+      var min = Infinity;
+      var next = -1;
+      var iter = 0;
+
+      for (var i = 0; i < graph.length; i++) {
+         if (visited[i] == false && distance[i] < min) {
+            next = i;
+            min = distance[i];
+         }
+      }
+      
+      if (next == -1) {
+         break;
+      }
+      current = next;
+      visited[current] = true;
+
+      for (var i = 0; i < graph[current].length; i++) {
```

Comment:
It looks like here you're iterating over the edges -- shouldn't that be part of your complexity?